modules/migrationmngr/README.md

```diff
@@ -0,0 +1,213 @@
+-   **If IBC is Not Enabled**: The migration can be performed "immediately" in a single block. All old validators are removed, and the new sequencer/attesters are added in one atomic update at the migration start height.
+
+### The Chain Halt: A Coordinated Upgrade
+
+The most critical part of the process is the final step. One block *after* the migration period ends (`migration_end_height + 1`), the `PreBlock` logic is designed to panic and halt the chain.
+
+This is an intentional design choice to force a coordinated upgrade. When the chain halts, node operators will see a specific error message in their logs:
+
+```
+MIGRATE: chain migration to evolve is complete. Switch to the evolve binary and run 'gmd evolve-migrate' to complete the migration.
+```
+
+This halt ensures that all participants must manually intervene to switch to the new software, preventing a chain split or other inconsistencies. Before halting, the module cleans up the migration state from the store to ensure the chain does not halt again on restart.
+
+## What Validators and Node Operators Need to Do
+
+If you are a validator or node operator on a chain using this module, you must be prepared for the migration event.
+
+1.  **Monitor Governance**: The migration will be initiated by a governance proposal. Stay informed about upcoming proposals. The proposal will define the target block height for the migration.
+
+2.  **Prepare for the Chain Halt**: Your node **will stop** at a predictable block height, calculated from the migration's start height (`block_height` in the proposal). With IBC enabled, the halt is at `block_height + 31`; without IBC, it is `block_height + 2`. This is expected. Check your node's logs for the specific "MIGRATE" error message.
+
+3.  **Perform the Upgrade**: Once the chain has halted, you must perform the following steps:
+    a. **Install the New Binary**: You will need to replace your current node software (e.g., `gmd`) with the new version required for the rollup.
+    b. **Run the Migration Command**: The error message will instruct you to run a command like `gmd evolve-migrate`. This command will handle any necessary state or configuration changes on your local node to make it compatible with the new network topology.
+    c. **Restart Your Node**: After running the migration command, you can restart your node. It will now be operating as part of the new rollup network (either as a sequencer or attester, depending on the new configuration).
+
+Your role will fundamentally change from a proof-of-stake validator to a participant in the rollup's new consensus mechanism.
+
+## Application Wiring (`app.go`)
+
+To ensure the `migrationmngr` module can correctly take control during a migration, you must make modifications to your application's main `app.go` file. This is crucial to prevent the standard `staking` module from sending conflicting validator set updates.
+
+### 1. Add the Migration Manager Keeper to the `App` Struct
```

Comment:
When adding the migration manager module, the staking module wrapper must as well be added.